src/graph/SelectionModel.ts

```diff
@@ -0,0 +1,51 @@
+export class SelectionModel {
```

Comment:
This could have some utility functions on it? (or maybe they'd live elsewhere, say on the Graph model.)

But things line select all/invert, include edges between - that sort of thing

Reply:
Yes it could. The renderer would need to be keyboard shortcuts / context menus to take advantage of them!